src/microbe_stage/player_microbe_control.cpp

```diff
@@ -215,12 +233,19 @@ void PlayerMicrobeControlSystem::Run(
     }
 }
 // ------------------------------------ //
-Float3 PlayerMicrobeControlSystem::getTargetPoint(Leviathan::GameWorld &worldWithCamera){
+Float3 PlayerMicrobeControlSystem::getTargetPoint(Leviathan::GameWorld &worldWithCamera, double targetAngle, ObjectID controlledEntity){
+
+	double x, y;
 
-    float x, y;
-    Engine::Get()->GetWindowEntity()->GetWindow()->GetNormalizedRelativeMouse(x, y);
+	Leviathan::Position& camera = worldWithCamera.GetComponent<Leviathan::Position>(controlledEntity);
+	Float3 camera_position = camera.Members._Position;
+	x = camera_position.GetX();
+	y = camera_position.GetY();
+	//So right now in order to avoid rounding errors I'm having to multiply by large numbers, presumably every order of magnitude of distance requires another 0, not ideal
+	x += 10000000 * std::cos(targetAngle);
+	y += 10000000 * std::sin(targetAngle);
```

Comment:
Maybe if you replace the ray casting (it is just to get the point on the x,z plane from a screen location) with just directly setting the target point somewhere near the cell in the right direction.

Reply:
Just thought of a way that might work. I'd have to have an invisible entity whose position on the screen is based on the target angle and some unit x to avoid rounding errors. Then I could treat it similar to how the mousepointer used to be treated. Does that sound feasible?

Reply:
Wouldn't it be better to just directly calculate the position of the invisible entity? It would be better to just have the target pos be playerCell.X + the cos math, playerCell.Z + the sin math and that would avoid the ray casting and be more efficient and probably less prone to rounding errors.

Reply:
Possibly, it would likely involve rewriting how movement is handled (I've just changed the inputs), because if the point the microbe is heading towards doesn't move as the microbe moves towards it, then you'll just get bouncing around a single point that moves in a circle.

Reply:
That's why I suggested adding the player position to the target to get it always be in front of the player cell in the right direction.

Reply:
So is it currently part of it? I think when I tried to do so it didn't quite work.